cherrymusicserver/__init__.py

```diff
@@ -171,24 +171,129 @@ def serve_file_utf8_fix(path, content_type=None, disposition=None,
 from cherrymusicserver import database
 from cherrymusicserver import httphandler
 from cherrymusicserver import log
+from cherrymusicserver import migrations
 from cherrymusicserver import playlistdb
 from cherrymusicserver import service
 from cherrymusicserver import sqlitecache
 from cherrymusicserver import userdb
 from cherrymusicserver import useroptiondb
 from cherrymusicserver import api
-import cherrymusicserver.browsersetup
 
 import audiotranscode
 MEDIA_MIMETYPES = audiotranscode.MIMETYPES.copy()
 del audiotranscode
 
+
+def setup_services():
+    """ services can be used by other parts of the program to easily access
+        different functions of cherrymusic by registering themselves as
+        service.user
+
+        See :mod:`~cherrymusicserver.services`.
+    """
+    service.provide('filecache', sqlitecache.SQLiteCache)
+    service.provide('cherrymodel', cherrymodel.CherryModel)
+    service.provide('playlist', playlistdb.PlaylistDB)
+    service.provide('users', userdb.UserDB)
+    service.provide('useroptions', useroptiondb.UserOptionDB)
+    service.provide('dbconnector', database.sql.SQLiteConnector, kwargs={
+        'datadir': pathprovider.databaseFilePath(''),
+        'extension': 'db',
+        'connargs': {'check_same_thread': False},
+    })
+
+
+def setup_config(override_dict=None):
+    """ Updates the internal configuration using the following hierarchy:
+        override_dict > file_config > default_config
+
+        Notifies the user if there are new or deprecated configuration keys.
+
+        See :mod:`~cherrymusicserver.configuration`.
+    """
+    defaults = cfg.from_defaults()
+    filecfg = cfg.from_configparser(pathprovider.configurationFile())
+    custom = defaults.replace(filecfg, on_error=log.e)
+    if override_dict:
+        custom = custom.replace(override_dict, on_error=log.e)
+    global config
```

Comment:
The config data the server uses is a module-level variable. Clearly not the best idea I ever had, but that's how I made it way back in history. 

Note that most of this code isn't new, but refactored from stuff that used to live in the CherryMusic class.